jdi-light-bootstrap-tests/src/test/java/io/github/epam/bootstrap/tests/composite/section/modal/ModalVerticallyCenteredTests.java

```diff
@@ -0,0 +1,168 @@
+                        modalVerticallyCentered.modal2
+                }
+        };
+    }
+
+    @DataProvider
+    public Object[][] modalCssData() {
+        return new Object[][]{
+                {
+                        modalVerticallyCentered.modalCenterTrigger,
+                        modalVerticallyCentered.dismissModal1Close,
+                        modalVerticallyCentered.modalCenterBg,
+                        "modal-dialog-centered"
+                },
+
+                {
+                        modalVerticallyCentered.modalCenterScrollableTrigger,
+                        modalVerticallyCentered.dismissModal2Close,
+                        modalVerticallyCentered.modalCenterScrollableBg,
+                        "modal-dialog-centered"
+                }
+        };
+    }
+
+    @DataProvider
+    public Object[][] modalVerticalAlignmentData() {
+        return new Object[][]{
+                {
+                        modalVerticallyCentered.modalCenterTrigger,
+                        modalVerticallyCentered.dismissModal1Close,
+                        "button:nth-of-type(1)",
+                        "modal-vertical-content-1"
+                },
+
+                {
+                        modalVerticallyCentered.modalCenterScrollableTrigger,
+                        modalVerticallyCentered.dismissModal2Close,
+                        "button:nth-of-type(2)",
+                        "modal-vertical-content-2"
+                }
+        };
+    }
+
+    @DataProvider
+    public Object[][] modalDismissData() {
+        return new Object[][]{
+                {
+                        modalVerticallyCentered.modalCenterTrigger,
+                        modalVerticallyCentered.dismissModal1Close,
+                        modalVerticallyCentered.closeX
+                },
+
+                {
+                        modalVerticallyCentered.modalCenterScrollableTrigger,
+                        modalVerticallyCentered.dismissModal2Close,
+                        modalVerticallyCentered.closeX
+                }
+        };
+    }
+
+    @BeforeMethod
+    public void before() {
+        shouldBeLoggedIn();
+        bsPage.shouldBeOpened();
+    }
+
+    @Test(dataProvider = "modalBasicData")
+    public void modalBasicFunctionalityTest(Button showButton,
+                                            Button dismissButton,
+                                            Modal modal) {
+        showButton.show();
+        showButton.click();
+
+        modal.is().displayed();
+
+        dismissButton.show();
+        dismissButton.click();
+
+        modal.is().hidden();
+    }
+
+    @Test(dataProvider = "modalCssData")
+    public void modalCssTest(Button showButton,
+                             Button dismissButton,
+                             Modal modal,
+                             String modalBgCss) {
+        showButton.show();
+        showButton.click();
+
+        modal.core().hasClass(modalBgCss);
```

Comment:
it's not a check. you need add is() here

Reply:
Done but there's a problem now.
I got a parent element I'm calling a **childs()** method upon.
When I'm trying to fetch the first child with **get(1)**, it returns the **second** child.
Is that really the way it's supposed to work at this point?

Reply:
possibly we should to fix childs method (as another task)

Reply:
Will be done during the meeting

Reply:
It self-resolved and now works as expected with get(1), get(2) etc...